packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/AccordianReferences.tsx

```diff
@@ -0,0 +1,139 @@
+
+import * as React from 'react';
+import cx from 'classnames';
+import IoIosArrowDown from 'react-icons/lib/io/ios-arrow-down';
+import IoIosArrowRight from 'react-icons/lib/io/ios-arrow-right';
+import { TNil } from '../../../../types';
+import './AccordianReferences.css';
+import { SpanReference } from '../../../../types/trace';
+import ReferenceLink from '../ReferenceLink';
+
+type AccordianReferencesProps = {
+  className?: string | TNil;
+  data: SpanReference[];
+  headerClassName?: string | TNil;
+  highContrast?: boolean;
+  interactive?: boolean;
+  isOpen: boolean;
+  label: React.ReactNode;
+  onToggle?: null | (() => void);
+  traceID: string;
+  focusSpan: (uiFind: string) => void;
+};
+
+type ReferenceItemProps = {
+  data: SpanReference[];
+  focusSpan: (uiFind: string) => void;
+  traceID: string;
+};
+
+// export for test
+export function References(props: ReferenceItemProps) {
+  const { data, focusSpan, traceID } = props;
+  return (
+    <div className="ReferencesList u-simple-scrollbars">
+      <ul className="ReferencesList--List">
+        {data.map(reference => {
+          return (
+            <li className="ReferencesList--Item" key={`${reference.spanID}`}>
+              <ReferenceLink reference={reference} traceID={traceID} focusSpan={focusSpan}>
+                {reference.span && reference.traceID === traceID && (
+                  <React.Fragment>
+                    <span className="span-svc-name"> {reference.span.process.serviceName} </span>
+                    <small className="endpoint-name">{reference.span.operationName} </small>
+                    <small className="SpanReference--debugInfo">
+                      <span className="SpanReference--debugLabel" data-label="Reference Type:">
+                        {reference.refType}
+                      </span>
+                      <span className="SpanReference--debugLabel" data-label="SpanID:">
+                        {reference.spanID}
+                      </span>
+                    </small>
+                  </React.Fragment>
+                )}
+                {reference.traceID !== traceID && (
+                  <React.Fragment>
+                    <span className="span-svc-name">&lt; span in another trace &gt;</span>
+                    <small className="SpanReference--debugInfo">
+                      <span className="SpanReference--debugLabel" data-label="Reference Type:">
+                        {reference.refType}
+                      </span>
+                      <span className="SpanReference--debugLabel" data-label="SpanID:">
+                        {reference.spanID}
+                      </span>
+                    </small>
+                  </React.Fragment>
+                )}
+              </ReferenceLink>
+            </li>
+          );
+        })}
+      </ul>
+    </div>
+  );
+}
+
+export default class AccordianReferences extends React.PureComponent<AccordianReferencesProps> {
```

Comment:
I totally agree with this, and I think this could be a separate PR, as this is getting too big.

I'm interested on working on it, so I can create the issue or if you create it can assign it to me if you think is appropriate.

Reply:
I cannot seem to assign it to you, so here is the issue: https://github.com/jaegertracing/jaeger-ui/issues/498